src/array/cpu/segment_reduce.cc

```diff
@@ -69,6 +75,14 @@ template void SegmentReduce<kDGLCPU, int64_t, double>(
     const std::string& op, NDArray feat, NDArray offsets, NDArray out,
     NDArray arg);
 
+template<> void ScatterAdd<kDGLCPU, int32_t, BFloat16>(
```

Comment:
Why do we need `<>` here?

Reply:
To partially specialize template, and avoid creating common template for this function for BFloat16

Reply:
I don't see difference better this ScatterAdd with the one on L80. Why they are different? Should we be consistent?